contracts/CloneFactory.sol

```diff
@@ -0,0 +1,46 @@
+pragma solidity ^0.4.18;
+
+/*
+The MIT License (MIT)
+
+Copyright (c) 2018 Murray Software, LLC.
+
+Permission is hereby granted, free of charge, to any person obtaining
+a copy of this software and associated documentation files (the
+"Software"), to deal in the Software without restriction, including
+without limitation the rights to use, copy, modify, merge, publish,
+distribute, sublicense, and/or sell copies of the Software, and to
+permit persons to whom the Software is furnished to do so, subject to
+the following conditions:
+
+The above copyright notice and this permission notice shall be included
+in all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
+CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+*/
+//solhint-disable max-line-length
+//solhint-disable no-inline-assembly
+
+contract CloneFactory {
+
+  event CloneCreated(address indexed target, address clone);
+
+  function createClone(address target) internal returns (address result) {
+    bytes memory clone = hex"34604457603a8060106000396000f30036600060ff376000803660ff73beefbeefbeefbeefbeefbeefbeefbeefbeefbeef5af48080156034573d80600060ff3e60fff3005b600080fd00";
```

Comment:
It'd be good for this to include a good comment which explains what each piece of this bytecode is and how it was derived so that it can be independently verified.

Reply:
Will open a new PR with updated bytecode and bytecode annotation. While doing the analysis I found extra `dup1` ops that are now removed from the upstream source.

Reply:
Might be good to write this in `lll` or something that does *pure* assembly so that you can have a dedicated *source* code version of this which is then used to generate this value and *template* it into the source code in some way.

Reply:
That's exactly how it's generated in the upstream [repo](https://github.com/yarrumretep/clone-factory). The full source file is downloaded when running `npm install` on this project and I copied the solidity contract to the top level to workaround compiler complaints. Do you think this is good enough for us?

Reply:
I still need to update this, but I did an opcode-by-opcode [analysis](https://gist.github.com/lsaether/d2dc85f3d0cf624b4ee56a631c68d0f1) so I'm comfortable enough with the source that writing it in LLL wouldn't be too hard. But I've never used LLL

Reply:
From what I've seen, LLL is a pretty thin layer on top of the opcodes themselves so conversion should be pretty trivial.  Whatever you get working is fine as long as it's reproduceable and auditable.